runtime/integration-tests/src/generic/utils/currency.rs

```diff
@@ -8,128 +8,158 @@ use cfg_types::{
 };
 use frame_support::{assert_ok, traits::OriginTrait};
 use sp_runtime::FixedPointNumber;
-use xcm::VersionedMultiLocation;
 
-use crate::generic::{config::Runtime, envs::fudge_env::FudgeSupport};
+use crate::generic::config::Runtime;
 
 pub const fn cfg(amount: Balance) -> Balance {
 	amount * CFG
 }
 
 pub trait CurrencyInfo {
-	const ID: CurrencyId;
-	const DECIMALS: u32;
-	const UNIT: Balance = 10u128.pow(Self::DECIMALS);
-	const SYMBOL: &'static str;
-	const NAME: &'static str = Self::SYMBOL;
-	const LOCATION: Option<xcm::VersionedMultiLocation> = None;
-	const CUSTOM: CustomMetadata;
-	const ED: Balance = 0;
-
-	fn metadata() -> AssetMetadata<Balance, CustomMetadata> {
+	fn id(&self) -> CurrencyId;
+
+	fn decimals(&self) -> u32 {
+		18
+	}
+
+	fn unit(&self) -> Balance {
+		10u128.pow(self.decimals())
+	}
+
+	fn symbol(&self) -> &'static str;
```

Comment:
Most of the time, we do not care about the symbol for testing. Could we use a default method here with some random/empty string? Maybe "Unnamed" or so.

Reply:
Agreed!